model/src/test/kotlin/utils/FindingsMatcherTest.kt

```diff
@@ -53,43 +56,67 @@ private fun createCopyrightFinding(statement: String, path: String, line: Int =
+
 class FindingsMatcherTest : WordSpec() {
     private val matcher = FindingsMatcher(LicenseFileMatcher("a/LICENSE"))
 
     init {
-        "getRootLicense()" should {
-            "return the license of the file matched by the license file matcher" {
+        "Given a license finding in a license file and a copyright finding in a file without license, match" should {
+            "associate that copyright with the root license" {
                 val licenseFindings = listOf(
-                    createLicenseFinding(license = "abc", path = "a/LICENSE")
+                    createLicenseFinding(license = "some id", path = "a/LICENSE")
+                )
+                val copyrightFindings = listOf(
+                    createCopyrightFinding(statement = "some stmt", path = "some/other/file")
                 )
 
-                matcher.getRootLicense(licenseFindings) shouldBe "abc"
+                val result = matcher.match(licenseFindings, copyrightFindings)
+
+                result.size shouldBe 1
+                val findings = result.getFindings("some id")
+                findings.locations.map { it.path } shouldContainExactlyInAnyOrder listOf("a/LICENSE")
+                findings.copyrights.map { it.statement } shouldContainExactlyInAnyOrder listOf("some stmt")
             }
+        }
 
-            "return an empty string when no file is matched by the license file matcher" {
-                val licenseFindings = listOf(
-                    createLicenseFinding(license = "abc", path = "b/LICENSE")
+        "Given no license finding in any license file and a copyright finding in a file without license, match" should {
+            "discard that copyright" {
+                val licenseFindings = listOf<LicenseFinding>()
+                val copyrightFindings = listOf(
+                    createCopyrightFinding(statement = "some stmt", path = "some/other/file")
                 )
 
-                matcher.getRootLicense(licenseFindings) shouldBe ""
+                val result = matcher.match(licenseFindings, copyrightFindings)
+
+                result.size shouldBe 0
             }
         }
 
-        "getClosestCopyrightStatements()" should {
-            "return exactly the statements within the line threshold" {
+        "Given a file with multiple license and copyright findings match" should {
+            "associate exactly the statements within the line threshold to the licenses and discard the others" {
                 // Use an arbitrary license start line that is clearly larger than DEFAULT_TOLERANCE_LINES.
                 val licenseStartLine = Random.nextInt(2 * DEFAULT_TOLERANCE_LINES, 20 * DEFAULT_TOLERANCE_LINES)
+                val licenseFindings = listOf(
+                    createLicenseFinding("license nearby", "path", licenseStartLine),
+                    createLicenseFinding("license far away", "path", licenseStartLine + 100 * DEFAULT_TOLERANCE_LINES)
```

Comment:
Out of scope of this change, but what would happen if you used `licenseStartLine + DEFAULT_TOLERANCE_LINES + 1` here, so that the copyright for "statement3" below is in the range of both licenses. Would it be associated to both, or the one that is closer, or randomly to one of them?

Reply:
A test in your last commit answers that question, it's both.